src/bin/ion/commands/dump.rs

```diff
@@ -104,39 +106,27 @@ pub(crate) fn run(_command: &str, args: &ArgMatches) -> Result<()> {
 
 /// Constructs the appropriate writer for the given format, then writes all values found in the
 /// Reader to the new Writer. If `count` is specified will write at most `count` values.
-pub(crate) fn write_in_format(
-    reader: &mut Reader,
+pub(crate) fn write_in_format<I: IonInput>(
+    reader: &mut Reader<AnyEncoding, I>,
     output: &mut Box<dyn Write>,
     format: &str,
     count: Option<usize>,
 ) -> IonResult<usize> {
-    // XXX: The text formats below each have additional logic to append a newline because the
-    //      ion-rs writer doesn't handle this automatically like it should.
-    //TODO: Solve these newline issues, get rid of hack
-    // https://github.com/amazon-ion/ion-cli/issues/36
-    // https://github.com/amazon-ion/ion-rust/issues/437
-    const NEWLINE: u8 = 0x0A;
     let written = match format {
         "pretty" => {
-            let mut writer = TextWriterBuilder::pretty().build(output)?;
-            let values_written = transcribe_n_values(reader, &mut writer, count)?;
-            writer.output_mut().write_all(&[NEWLINE])?;
-            Ok(values_written)
+            let mut writer = Writer::new(v1_0::Text.with_format(TextFormat::Pretty), output)?;
+            transcribe_n_values(reader, &mut writer, count)
```

Comment:
Did the newline issue get solved some other way? 
https://github.com/amazon-ion/ion-rust/issues/437 is still open

Reply:
I closed it just now; the new `Writer` that replaced the `IonWriter` trait does not have the newline issue.